storage/badger/headers.go

```diff
@@ -29,6 +29,7 @@ func NewHeaders(collector module.CacheMetrics, db *badger.DB) *Headers {
 		return operation.InsertHeader(blockID, header)
 	}
 
+	// this method should only be used to index the height of finalized blocks
```

Comment:
thanks for adding the comment. 

```suggestion
	// CAUTION: should only be used to index FINALIZED blocks by their respective height
```